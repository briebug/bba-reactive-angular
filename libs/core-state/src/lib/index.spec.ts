import { mockLesson, mockCourse } from "./tests.mocks";
import { getCourseLessons } from '.';

describe('Index Common Selector', () => {
    let lessons, courses;

    beforeEach(() => {
        lessons = [
            {...mockLesson, id: '0'},
            {...mockLesson, id: '1'},
            {...mockLesson, id: '2'},
        ]
        courses = [
            {...mockCourse, id: '0'},
            {...mockCourse, id: '1'},
            {...mockCourse, id: '2'},
        ]
    })


    it('should select the related courses and lessons', () => {
        expect(getCourseLessons.projector(courses, lessons)).toMatchSnapshot();
    })
})