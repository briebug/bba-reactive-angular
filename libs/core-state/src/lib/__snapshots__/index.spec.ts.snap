// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Common Selector should select the related courses and lessons 1`] = `
Array [
  Object {
    "description": "mockCourse constant for testing",
    "id": "0",
    "lessons": Array [
      Object {
        "course_id": "0",
        "description": "mockLesson constant for testing",
        "id": "0",
        "title": "mockLesson",
      },
      Object {
        "course_id": "0",
        "description": "mockLesson constant for testing",
        "id": "1",
        "title": "mockLesson",
      },
      Object {
        "course_id": "0",
        "description": "mockLesson constant for testing",
        "id": "2",
        "title": "mockLesson",
      },
    ],
    "title": "mockCourse",
  },
  Object {
    "description": "mockCourse constant for testing",
    "id": "1",
    "lessons": Array [],
    "title": "mockCourse",
  },
  Object {
    "description": "mockCourse constant for testing",
    "id": "2",
    "lessons": Array [],
    "title": "mockCourse",
  },
]
`;
